llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -1387,11 +1455,20 @@ bool GCNPassConfig::addRegAssignAndRewriteFast() {
 
   // Equivalent of PEI for SGPRs.
   addPass(&SILowerSGPRSpillsID);
+
+  // To Allocate wwm registers used in whole quad mode operations (for pixel
+  // shaders).
   addPass(&SIPreAllocateWWMRegsID);
```

Comment:
I think the comment in the code is misleading since WWM registers pre-allocated are used in all shader types (not just PS) for atomic optimization and subgroup intercommunication.

Reply:
I will mention the shaders instead of PS alone.

Reply:
> My comment here maybe should be a follow up to this patch: SIPreAllocateWWMRegs perhaps should be renamed to SIPreAllocateWQMRegsInPS, and its functionality should be limited to WQM registers. 

I would rather not touch it in this patch. These passes will anyway get a revamp in your WIP allocation patches. Let's fix the pass name at that point.